src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Types/TypeFactory.cs

```diff
@@ -21,8 +20,6 @@ internal sealed class TypeFactory
             type.outerType = outerType;
             type.SetOwningAggregate(parent);
             type.SetTypeArgsThis(typeArgsThis);
-            type.SetName(name);
```

Comment:
`GetName()` is never actually used on any type except for `TypeParameterType` and will be refactored out of here. For the purposes of this PR, since we've removed the mechanism for building that name, we can just leave it as null.